File: VB/ExtendModel.Module.Web/WebGroupFooterViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Data
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Web.Editors.ASPx
Imports DevExpress.Web

Namespace ExtendModel.Module.Web
	Public Class WebGroupFooterViewController
		Inherits ViewController(Of ListView)
		Protected Overrides Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()
			Dim modelListView As IModelListViewExtender = TryCast(View.Model, IModelListViewExtender)
			If modelListView IsNot Nothing AndAlso modelListView.IsGroupFooterVisible Then
				Dim gridListEditor As ASPxGridListEditor = TryCast(View.Editor, ASPxGridListEditor)
				If gridListEditor IsNot Nothing Then
					Dim gridView As ASPxGridView = gridListEditor.Grid
					gridView.Settings.ShowGroupFooter = GridViewGroupFooterMode.VisibleAlways
					For Each modelColumn As IModelColumn In View.Model.Columns
						Dim modelColumnExtender As IModelColumnExtender = TryCast(modelColumn, IModelColumnExtender)
						If modelColumnExtender IsNot Nothing AndAlso modelColumnExtender.GroupFooterSummaryType <> SummaryItemType.None Then
							Dim fieldName As String = modelColumn.ModelMember.MemberInfo.BindingName
							Dim summaryItem As ASPxSummaryItem = Nothing
							For Each currentItem As ASPxSummaryItem In gridView.GroupSummary
								If currentItem.FieldName = fieldName Then
									currentItem.ShowInGroupFooterColumn = modelColumn.Caption
									summaryItem = currentItem
									Exit For
								End If
							Next currentItem
							If summaryItem Is Nothing Then
								summaryItem = New ASPxSummaryItem(fieldName, modelColumnExtender.GroupFooterSummaryType)
								summaryItem.ShowInGroupFooterColumn = modelColumn.Caption
								gridView.GroupSummary.Add(summaryItem)
							End If
						End If
					Next modelColumn
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/ExtendModel.Module.Web/Controllers/WebGroupFooterViewController.cs
using System;
using DevExpress.Data;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Web.Editors.ASPx;
using DevExpress.Web;

namespace ExtendModel.Module.Web.Controllers {
    public class WebGroupFooterViewController : ViewController<ListView> {
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            IModelListViewExtender modelListView = View.Model as IModelListViewExtender;
            if (modelListView != null && modelListView.IsGroupFooterVisible) {
                ASPxGridListEditor gridListEditor = View.Editor as ASPxGridListEditor;
                if (gridListEditor != null) {
                    ASPxGridView gridView = gridListEditor.Grid;
                    gridView.Settings.ShowGroupFooter = GridViewGroupFooterMode.VisibleAlways;
                    foreach (IModelColumn modelColumn in View.Model.Columns) {
                        IModelColumnExtender modelColumnExtender = modelColumn as IModelColumnExtender;
                        if (modelColumnExtender != null &&
                            modelColumnExtender.GroupFooterSummaryType != SummaryItemType.None) {
                            string fieldName = modelColumn.ModelMember.MemberInfo.BindingName.ToUpper();
                            ASPxSummaryItem summaryItem = null;
                            foreach (ASPxSummaryItem currentItem in gridView.GroupSummary) {
                                if (currentItem.FieldName == fieldName) {
                                    currentItem.ShowInGroupFooterColumn = modelColumn.Caption.ToUpper();
                                    summaryItem = currentItem;
                                    break;
                                }
                            }
                            if (summaryItem == null) {
                                summaryItem = new ASPxSummaryItem(
                                    fieldName, modelColumnExtender.GroupFooterSummaryType);
                                summaryItem.ShowInGroupFooterColumn = modelColumn.Caption.ToUpper();
                                gridView.GroupSummary.Add(summaryItem);
                            }
                        }
                    }
                }
            }
        }
    }
}
